added event listerner for open and close modal action

pragyes31 opened the pull request.

c-ciobanu commented on src/Component/HomePage.js:
```diff
@@ -1,14 +1,35 @@
+    }
+      handleModalWindow = e => {
+        e.preventDefault();
+        this.setState(() => ({ isModalOpen: !this.state.isModalOpen }));
```
For a better setState for this case follow [this guide ](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous).

pragyes31 replied:
thanks for the doc. The only thing differently I did here is passing state as an argument in the callback function inside setState.

c-ciobanu commented on src/Component/HomePage.js:
```diff
@@ -1,14 +1,35 @@
+                <SignUpModal 
+                openModal={this.state.isModalOpen} 
+                closeModal={this.handleModalWindow} />
+                <SignInModal   
```
The 2 modals depend on the same state so once you open one of them the other one will open too.
Is better to have separate states for each modal.

pragyes31 replied:
That's a great Catch Cristi!! Added separate state for both.

pragyes31 commented:
@c-ciobanu I added separate state for both the modal windows, indented the Homepage code and added prop-types lib to our project.

c-ciobanu commented:
@pragyes31 All good except the propTypes definition that should stay inside the signup and signin components instead of the homepage, you should define prop types only if the components uses props, in this case the homepage doesn't have props.

pragyes31 commented:
@c-ciobanu you're right. I added the prop type in the modal components.

c-ciobanu commented:
@pragyes31 The closeModal propType should be of type func, except for this I think we are good and can merge this PR.

pragyes31 commented:
@c-ciobanu I added the correct prop type. Also added handleSwapModal function to HomePage.js which would enable the user to switch between sign-in and sign-up modal.
This is when let's say the user is in sigh-in modal but wishes to sign-up instead and vice versa.
